src/logic/tokens/store/actions/fetchSafeTokens.ts

```diff
@@ -24,17 +29,20 @@ interface ExtractedData {
   tokens: List<Token>
 }
 
-const extractDataFromResult = (currentTokens: TokenState) => (
+const { nativeCoin } = getNetworkInfo()
+
+const extractDataFromResult = (currentTokens: TokenState, fiatCode: string) => (
   acc: ExtractedData,
-  { balance, fiatBalance, fiatCode, token, tokenAddress }: BalanceEndpoint,
+  { balance, fiatBalance, tokenInfo }: TokenBalance,
 ): ExtractedData => {
-  if (tokenAddress === null) {
+  const { address: tokenAddress, decimals } = tokenInfo
+  if (sameAddress(tokenAddress, ZERO_ADDRESS) || sameAddress(tokenAddress, nativeCoin.address)) {
```

Comment:
Note that the transaction services returns the balance for the native token with `null` address and the client gateway returns it as `0x0000000000000000000000000000000000000000` that's why I needed this change